Sources/HaCWebsiteLib/Events/EventServer.swift

```diff
@@ -2,22 +2,25 @@ import Foundation
 
 public struct EventServer {
 
-  static func getEvents() -> [GeneralEvent] {
-    //TODO: source events from database
-    let hypeDuration : Double = 4000000
-    let duration : Double = 4000
-    let workshopEvents : [GeneralEvent] = WorkshopManager.workshops.map{
-      (workshop: Workshop) -> GeneralEvent in
-        return GeneralEvent(title: workshop.title, time:  DateInterval(start: Date(), duration: duration),
-          tagLine: "There's Pizza", color: "purple", hypePeriod: DateInterval(start: Date(), duration: hypeDuration),
-          tags: ["tag1", "tag2"], description: Markdown("Thing"))
-    }.filter{ event in 
-      event.shouldShowAsUpdate
+  private(set) static var events: [GeneralEvent] = []
+
+  static func update() {
+    let newEvents = try? GeneralEvent.makeQuery().filter("title", .notEquals, nil).all()
+    guard let notNilEvents = newEvents else {
+      events = []
+      return
+    }
+    events = notNilEvents
+  }
+
+  static func getAllEvents() -> [GeneralEvent] {
+    return events
+  }
+
+  static func getCurrentEvents() -> [GeneralEvent] {
+    return events.filter{ event in 
+      let currentDate = Date()
```

Comment:
Why are we doing this? Surely plan is to show all future events on the landing page. Also, surely doing this through a database search is faster that filtering through an in-memory array?

Reply:
Well, we may not wish to start advertising for all future events, just the ones which are currently being pushed. I think that this probably goes best alongside a `getCurrentAndFutureEvents` function though

Reply:
I mean, I thought the plan is to eventually show a general `event-stream` on the landing page, grabbing information from all our content: blog-posts, workshops, hackathons, videos, some shoutouts etc. Possibly going into the past and future.

Reply:
On the landing page we will want to keep the number of things we display to a minimum so this will only be current events. Sure we may have past videos/blog posts but that's a completely different matter of `updates` not `events.